source/common/access_log/access_log_impl.cc

```diff
@@ -81,8 +83,15 @@ bool StatusCodeFilter::evaluate(const RequestInfo::RequestInfo& info, const Http
 }
 
 bool DurationFilter::evaluate(const RequestInfo::RequestInfo& info, const Http::HeaderMap&) {
+  Optional<std::chrono::nanoseconds> final = info.finalTimeMonotonic();
+
+  if (!final.valid()) {
```

Comment:
I definitely think we should do error handling here but should we consider not finalizing valid or a bug?  It's especially weird that if you don't finalize, every time you evaluate you might get a different result.

Reply:
So in an earlier commit I rolled back (before PR) I had assertions to make sure finalTimeMonotonic wasn't called until it was set (and wasn't an optional value) but that seems like a weird way for this code to fail. I could throw instead? Not sure what makes the most sense.

Reply:
I would consider it a bug and just do `ASSERT`. It's a programming error that we should be catching in tests.